AzureMonitorAgent/agent.py

```diff
@@ -395,14 +415,192 @@ def update():
     
     return 0, ""
 
+def stop_metrics_process():
+
+    #Stop the telegraf and ME services
+    tel_out, tel_msg = telhandler.stop_telegraf_service(is_lad=False)
+    if tel_out:
+        HUtilObject.log(tel_msg)
+    else:
+        HUtilObject.error(tel_msg)
+
+    me_out, me_msg = me_handler.stop_metrics_service(is_lad=False)
+    if me_out:
+        HUtilObject.log(me_msg)
+    else:
+        HUtilObject.error(me_msg)
+
+    #Delete the telegraf and ME services
+    tel_rm_out, tel_rm_msg = telhandler.remove_telegraf_service()
+    if tel_rm_out:
+        HUtilObject.log(tel_rm_msg)
+    else:
+        HUtilObject.error(tel_rm_msg)
+
+    me_rm_out, me_rm_msg = me_handler.remove_metrics_service(is_lad=False)
+    if me_rm_out:
+        HUtilObject.log(me_rm_msg)
+    else:
+        HUtilObject.error(me_rm_msg)
+
+    pids_filepath = os.path.join(os.getcwd(),'amametrics.pid')
+
+    # kill existing telemetry watcher
+    if os.path.exists(pids_filepath):
+        with open(pids_filepath, "r") as f:
+            for pids in f.readlines():
+                kill_cmd = "kill " + pids
+                run_command_and_log(kill_cmd)
+                run_command_and_log("rm "+pids_filepath)
+
+def start_metrics_process():
+    """
+    Start telemetry process that performs periodic monitoring activities
+    :return: None
+
+    """
+    stop_metrics_process()
 
+    package_directory = os.path.join(os.getcwd(), PackagesDirectory)
+    telegraf_path = os.path.join(package_directory, TelegrafBinName)
```

Comment:
Telegraf will also be part of the latest mdsd deb/rpm and it will copy it to /usr/sbin/telgraf as it does with mdsd.

Reply:
as discussed, its not there in the buidl yet, so we will remove this once its done